### ISSUE TITLE
BlenderBIM: drawings coming up empty. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#1504]
Might be doing something wrong, but [this](https://gitlab.com/openingdesign/boathouse_monona/-/tree/972da163261b91046bc782c11ee307270c57a74c/Models%20and%20CAD) drawing comes up empty. 

Video: https://www.dropbox.com/s/h2i217i210i8nfj/2021-05-28_23-20-25.mp4?dl=0

**Comment by Moult** [comment]
The new drawing system is far from ready - currently for example it only supports one drawing in a file. Maybe hold off testing drawings for a while. Unfortunately not much I can do right now on my end.

**Comment by theoryshaw** [comment]
Sounds good.

**Comment by RickBrice** [comment]
I've generated some IFC 4x2 and 4x3 files from my software with bridge models and linear referencing. Nothing shows up in Blender BIM. Is this supported?

**Comment by theoryshaw** [comment]
Hi @RickBrice You might want to ask your question here: https://community.osarch.org/categories/general

Github issues, is good for specific bugs or feature requests.

**Comment by aothms** [comment]
@RickBrice not really yet, together with @civilx64 we're dipping our toes in this. See #842 #1470 #1480

**Comment by civilx64** [comment]
@RickBrice out of curiosity which software are you utilizing to generate the files?  I ask because we were planning to support schema versions 4.1 and 4.3 but not necessarily 4.2 because it was withdrawn before becoming an official standard. 

This may not be a major issue, at least for alignments, because I don't believe there were significant changes between 4.1 and 4.2. The bridge entities are a different story - although they have not received much focus yet until we have tested alignments more fully.

**Comment by RickBrice** [comment]
@civilx64 I am developing a plug-in for a program called [BridgeLink:PGSuper](https://github.com/WSDOT/PGSuper). The program doesn't currently have IFC BIM capabilities but it has quite a bit of detailed information about precast/prestressed concrete bridge superstructures including parametric geometry based on alignment information, girder framing, deck geometry, etc. 

I'm pretty new to IFC. I'm unsure of the 3D solid model information I'm exporting and hoping to find a viewing tool to help me learn and validate exports.

**Comment by aothms** [comment]
Hi @RickBrice I just chatted with one of the main 4.3 developers. He suggested:

> I would propose to ignore IFC4x2 as it never became an official version.
> In IFC4x3 we did a rather large architectural change to the alignment
> and related geometry.

If you drop me an email at thomas@ifcopenshell.org I can introduce you to the group. They'll likely have some tools to help.

**Comment by RickBrice** [comment]
@aothms - I would agree that skipping 4x2 and focusing on 4x3 is the right move. I've been using what's available in IfcOpenShell to get me started. I generated code for the 4x3 RC3 and RC4 schemas and added them to the IfcParse library. I'm focusing on RC3 for now because it seems like there might be some tools that support it. The changes to RC4 are pretty small, at least for Alignment and Linear Placement.

**Comment by civilx64** [comment]
> hoping to find a viewing tool to help me learn and validate exports.

I have found BlenderBIM to be very useful in this regard.  For example, last fall I was modeling some ground-mounted roadway signs with IFC using [IfcRightCircularCylinder](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifcgeometricmodelresource/lexical/ifcrightcircularcylinder.htm) for the footing and extruding [IfcIShapeProfileDef](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifcprofileresource/lexical/ifcishapeprofiledef.htm)s to represent the W12 posts.  Importing to Microstation would drop the cylindrical footings entirely and not include the fillets in the posts.  Everything came in just fine in BlenderBIM.  I also used IfcConvert to export to glTF so we could texture the model with a jpg of the sign graphics... although I could have done that in Blender or even added the texture into the IFC data.
